internal/datacoord/channel.go

```diff
@@ -39,7 +44,30 @@ type RWChannel interface {
 	UpdateWatchInfo(info *datapb.ChannelWatchInfo)
 }
 
-var _ RWChannel = (*channelMeta)(nil)
+func NewRWChannel(name string,
+	collectionID int64,
+	startPos []*commonpb.KeyDataPair,
+	schema *schemapb.CollectionSchema,
+	createTs uint64,
+) RWChannel {
+	if paramtable.Get().DataCoordCfg.EnableBalanceChannelWithRPC.GetAsBool() {
+		return &StateChannel{
```

Comment:
This StateChannel looks invalid for missing currentState and assignedNode, recommand to set currentState="standby" and assignedNode=bufferID